addon-test-support/@ember/test-helpers/dom/type-in.ts

```diff
@@ -46,27 +46,30 @@ export default function typeIn(target: Target, text: string, options: Options =
       throw new Error('Must pass an element or selector to `typeIn`.');
     }
 
-    const element = getElement(target);
+    const element = getElement(target) as Element | HTMLElement;
```

Comment:
Partially because that's what `fillIn` was doing. There was a type error that I don't recall. I'll go check on it.

Reply:
kk, doesn't seem like a major issue to me, but if we have a type misalignment in `getElement` we can try to fix it instead of forcing all consumers to cast. Alternatively, if specific helper methods can _only_ work on `HTMLElement` (vs the more generic `Element`) maybe we should make an assertion method that narrows the types for us.

Reply:
I pushed a commit to this branch that alters the type guards and removes the casting but it's not reflecting in the PR 🤷

Reply:
One bit to note here about how both fillIn and typeIn work is they're assuming FormControl is ok, but that includes selects and buttons which don't seem like they should be allowed. Might be nice to have some sort of fillable type guard that narrows down to input/textarea and maybe content editable?

Reply:
Yeah, totally agree. Seems like a great candidate for a follow up PR if you have the time.